ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java

```diff
@@ -65,6 +67,10 @@ static <T> void handleOldApiRequest(
         return;
       }
       requestObject = optRequestObject.get();
+      if (requestObject instanceof KsqlRequest) {
+        final KsqlRequest request = (KsqlRequest) requestObject;
+        request.setMaskedKsql(QueryMask.getMaskedStatement(request.getKsql()));
```

Comment:
How come the masking takes place here rather than in the constructor for KsqlRequest?

Reply:
KsqlRequest is in `ksqldb-rest-model` module and `QueryMask` is in `ksqldb-parser` and needs dependency in `ksqldb-parser`.

`ksqldb-parser` depends on `ksqldb-common` and `ksqldb-metastore`.
`ksqldb-rest-model` depends on `ksqldb-common`, `ksqldb-engine-common
`ksqldb-engine-common` depends on `ksqldb-common`

If I add dependencies of `ksqldb-parser` in `ksqldb-rest-model`, it creates circular dependencies:

`ksqldb-rest-model` -> `ksqldb-parser` -> `ksqldb-metastore` -> `ksqldb-execution` -> `ksqldb-rest-model`

Reply:
I see, that's unfortunate. It means we have to catch all the places where KsqlRequest may be created, of which there are multiple... Almost makes me wonder if we should just use a regex match instead to remove the parser dependency. The regex would be more complicated, though, since we'd want to avoid masking CREATE STREAM/TABLE statements as much as possible. Maybe it's not worth the complexity.

Reply:
Maybe we can create `MaskedKsqlRequest` in other module which wraps KsqlRequest and replace where `KsqlRequest` is used with masked one. That will be a larger change I can do later if needed I guess... For now, will just try best to set the masked one for `KsqlRequest` and throw exception in `getMaskedSql` if not set and use `ksql` in `toString` if `maskedSql` is not set